### PR TITLE
Speed up point_to_coords

### PR DESCRIPTION
**Pull request by kngwyu**
- if cache doesn't exist, find coord when making cache
- else, use binary search

### CODE REVIEW

File: src/racer/core.rs [hunk 1]
```diff
@@ -557,11 +575,21 @@ impl IndexedSource {
...
-        // TODO: binary search is better?
-        for (i, &range) in self.lines.borrow().iter().enumerate() {
-            if range.contains(point) {
-                return Some(Coordinate::new(i as u32 + 1, (point - range.start).0 as u32))
+        if let Some(cd) = self.cache_lineoffsets_and_get_line(point) {
+            return Some(cd);
+        }
+        let lines = self.lines.borrow();
+        // find appropriate coord using binary search
+        let (mut lo, mut hi) = (0, lines.len());
```
Comment (birkenfeld) [inline src/racer/core.rs]: Could this use `slice::binary_search_by`?

Comment (kngwyu) [reply src/racer/core.rs]: Sounds good:)

**Comment by kngwyu** [comment]
Now it looks much smarter, thanks @birkenfeld 

Here's the result of std benchmark:
before
```
running 3 tests
test completes_hashmap               ... bench:     798,103 ns/iter (+/- 43,783)
test completes_methods_for_file_open ... bench:     772,886 ns/iter (+/- 32,676)
test completes_methods_for_vec       ... bench: 112,440,661 ns/iter (+/- 5,331,488)
```
after
```
running 3 tests
test completes_hashmap               ... bench:     808,920 ns/iter (+/- 25,661)
test completes_methods_for_file_open ... bench:     786,085 ns/iter (+/- 45,139)
test completes_methods_for_vec       ... bench: 109,354,725 ns/iter (+/- 1,658,401)
```

**Comment by birkenfeld** [comment]
Doesn't look much faster...?

**Comment by kngwyu** [comment]
Yeah, but I don't know why :confused:

**Pull request merged**